src/dialog.rs

```diff
@@ -0,0 +1,261 @@
+//! User interaction with the file system using dialog boxes.
+//!
+//! # Example
+//!
+//! ```rust,no_run
+//! use tauri_api::dialog::open;
+//!
+//! let path = open.await;
+//! ```
+use serde::Serialize;
+use std::path::PathBuf;
+
+/// Extension filter for the file dialog.
+///
+/// # Example
+///
+/// ```rust,no_run
+/// let filter = DialogFilter {
+///   extension: vec![".jpg", ".jpeg", ".png", ".bmp"],
+///   name: "images",
+/// };
+/// ```
+#[derive(Serialize)]
+pub struct DialogFilter {
+    /// Extensions to filter, without a `.` prefix.
+    pub extensions: Vec<String>,
+
+    /// Filter name
+    pub name: String,
+}
+
+/// Types of a [`message`] dialog.
+#[derive(Serialize)]
+pub enum MessageDialogType {
+    Error,
+    Info,
+    Warning,
+}
+
+/// Options for the [`message`] dialog.
+#[derive(Serialize)]
+pub struct MessageDialogOptions {
+    /// The title of the dialog. Defaults to the app name.
+    pub title: Option<String>,
+
+    /// The type of the dialog. Defaults to MessageDialogType::Info.
+    pub kind: MessageDialogType,
```

Comment:
In [the TS spec](https://github.com/tauri-apps/tauri/blob/e4292ce7be198f6265dde9b6ae54e73b7c41e7da/tooling/api/src/dialog.ts#L96) this field is called `type`. I changed it to `kind` to avoid a collision with the `type` keyword in Rust. Another option would be to call it `type_`, as they do in `web_sys` ([e.g.](https://rustwasm.github.io/wasm-bindgen/api/web_sys/struct.MediaQueryList.html#method.remove_event_listener_with_callback)).

Reply:
I think you can also do `r#type`, but I think `kind` is fine too